upstream/remote/remote.go

```diff
@@ -188,10 +190,12 @@ func (r *Remote) uploadProfile(j *upstream.UploadJob) error {
 	request.Header.Set("Content-Type", contentType)
 	// request.Header.Set("Content-Type", "binary/octet-stream+"+string(j.Format))
 
-	if r.cfg.AuthToken != "" {
+	if r.cfg.AuthToken != "" && isOGPyroscopeCloud(u) {
 		request.Header.Set("Authorization", "Bearer "+r.cfg.AuthToken)
 	} else if r.cfg.BasicAuthUser != "" && r.cfg.BasicAuthPassword != "" {
 		request.SetBasicAuth(r.cfg.BasicAuthUser, r.cfg.BasicAuthPassword)
+	} else if r.cfg.AuthToken != "" {
+		r.logger.Infof("Deprecated: AuthToken is being used. Please switch to BasicAuthUser and BasicAuthPassword.")
```

Comment:
>  AuthToken is being used
It says the token is beeing used, but in fact it is ignored. Maybe say something like "Authtoken is specified, but deprecated and ignored Please switch to ..."

Also if someone still wants to use Bearer (for example they put private pyroscope behind nginx with auth), they can use `HTTPHeaders `, maybe it is also worth mentioning in the warning.

Reply:
Btw, I set it as `Info` not `Warn` since `Logger` [interface](https://github.com/grafana/pyroscope-go/blob/main/upstream/remote/remote.go#L55) doesn't implement it but I could if needed, lmkwyt.

Reply:
I think it is fine.  I think if we add a a method to logger  interface it will break compilation for all users who has custom logger implementation